lms/djangoapps/courseware/views/views.py

```diff
@@ -449,15 +450,22 @@ def register_user_access_warning_messages(request, course_key):
                 )
             )
         elif not is_enrolled and not is_staff:
-            PageLevelMessages.register_warning_message(
-                request,
-                Text(_('You must be enrolled in the course to see course content. {enroll_link}.')).format(
-                    enroll_link=HTML('<a href="{url_to_enroll}">{enroll_link_label}</a>').format(
-                        url_to_enroll=CourseTabView.url_to_enroll(course_key),
-                        enroll_link_label=_("Enroll now"),
+            # Only show enroll button if course is open for enrollment.
+            if course_open_for_self_enrollment(course_key):
+                enroll_message = _('You must be enrolled in the course to see course content. \
+                        {enroll_link_start}Enroll now{enroll_link_end}.')
+                PageLevelMessages.register_warning_message(
+                    request,
+                    Text(enroll_message).format(
+                        enroll_link_start=HTML('<button class="enroll-btn btn-link">'),
+                        enroll_link_end=HTML('</button>')
                     )
                 )
-            )
+            else:
+                PageLevelMessages.register_warning_message(
+                    request,
+                    Text(_('You must be enrolled in the course to see course content.'))
```

Comment:
I tried it without `Text` and the button html gets html-escaped, so that's why it's there.